### DISCUSSION TITLE
Halfmoon Previewer ( Visual Studio Code Extension )

### DISCUSSION THREAD

**Discussion by Supernova3339** [halfmoonui/halfmoon#138] [Show and tell]
Hey there, developers!

I'm excited to introduce the Halfmoon Previewer extension for Visual Studio Code. 🎉 

## Features

- **Preview HalfmoonUI Projects:** View your HalfmoonUI projects directly within Visual Studio Code.
- **Real-time Updates:** Experience real-time updates to the preview as you edit your HalfmoonUI code.
- **Light and Dark Themes:** Choose between light and dark themes to match your project's design.

## Installation

1. Open Visual Studio Code.
2. Go to the Extensions view by clicking on the square icon in the sidebar.
3. Search for "Halfmoon Previewer" and click Install.

## Usage

1. Open your HalfmoonUI project in Visual Studio Code.
2. Open a HalfmoonUI file in the editor.
3. Run the "Open Previewer" command using the command palette. (`Ctrl+Shift+P` or `Cmd+Shift+P` on macOS)

## How It Works

The Halfmoon Previewer extension creates a WebView panel that renders the HalfmoonUI code you're working on. As you make changes to your code, the preview automatically updates to reflect those changes.

## Configuration

You can configure the extension to use either the light or dark theme for the preview. The theme is determined based on your active color theme in Visual Studio Code.

## Feedback and Contributions

Have suggestions or found a bug? Please [open an issue](https://github.com/supernova3339/halfmoon-previewer/issues) on our GitHub.

Contributions are welcome! If you'd like to contribute, please fork the repository and create a pull request.

## License

This extension is licensed under the [MIT License](https://github.com/Supernova3339/halfmoon-previewer/blob/master/LICENSE.md).

---

If you find this extension useful, consider [sponsoring me](https://github.com/sponsors/Supernova3339/) to support ongoing development.

🔗 [Get Halfmoon Previewer from the Visual Studio Code Marketplace](https://marketplace.visualstudio.com/items?itemName=Supernova3339.halfmoon-previewer)

Looking forward to your feedback and contributions! Let's make HalfmoonUI development even more enjoyable. 🌙

**Comment by Art9681** [comment]
Working great so far. Thank you!

**Reply by Supernova3339** [reply]
Thanks for your feedback!

**Comment by HarshNarayanJha** [comment]
Sounds Great. I will it try soon

**Reply by Supernova3339** [reply]
Thank you!
